tests/test_cli.py

```diff
@@ -15,25 +15,23 @@
 import rsa
 import rsa.cli
 import rsa.util
+from rsa._compat import PY3
 
-if sys.version_info[0] < 3:
-    def make_buffer():
-        return BytesIO()
 
-
-    def get_bytes_out(out):
-        # Python 2.x writes 'str' to stdout:
-        return out.getvalue()
-else:
-    def make_buffer():
+def make_buffer():
```

Comment:
Hmmmm now that I see how it turns out, I have some doubts about doing the `if PY3` inside the function. The way it is now, `if PY3` is evaluated for each and every function call. By doing something like this:

```python
if PY3:
    def write_to_stdout(data): ...
    def make_buffer(): ...
else:
    def write_to_stdout(data): ...
    def make_buffer(): ...
```

we only need to evaluate `if PY3` once. This is slightly harder to read, as the PY3 and PY2 versions of the functions are potentially further apart, it does make the runtime execution a bit faster. What do you think @adamantike ?